lib/puppet/provider/base_dsc/powershell.rb

```diff
@@ -15,6 +17,18 @@
 Applies DSC Resources by generating a configuration file and applying it.
 EOT
 
+  def initialize(value={})
+    super(value)
+    set_symlink_folder
+  end
+
+  def set_symlink_folder
+    puppetModulesVendorFolder = "#{ENV['ProgramFiles']}\\WindowsPowerShell\\Modules\\PuppetVendoredModules"
```

Comment:
So a question about this that maybe I should have raised before in the old version... we're making a little bit of an assumption here about location (especially since we're not even looking at `PSModulePath`)

* Do we care about which module path we use since there are a few?  I would think that we want to prefer `Machine` level settings to `User`?
* Are we sure that that module path is always injected for all users (i.e. for `LocalSystem`)?
* Do we want to insulate ourselves from env vars by using the registry (or calling `GetEnvironmentVariable`)?

I did a quick test with my local Administrator:

```
> $env:PSModulePath -split ';'

C:\Users\Administrator\Documents\WindowsPowerShell\Modules
C:\Program Files\WindowsPowerShell\Modules
C:\Windows\system32\WindowsPowerShell\v1.0\Modules\
C:\Program Files (x86)\Microsoft SDKs\Windows Azure\PowerShell\Azure
```

And also with `SYSTEM` (that shows we still get the `Program Files`):

```
PS C:\Windows\system32> $env:PSModulePath -split ';'

WindowsPowerShell\Modules
C:\Program Files\WindowsPowerShell\Modules
C:\Windows\system32\WindowsPowerShell\v1.0\Modules\
C:\Program Files (x86)\Microsoft SDKs\Windows Azure\PowerShell\Azure
```

But note that the `C:\Program Files\WindowsPowerShell\Modules` isn't a configurable `PSModulePath`, so must otherwise be injected.

```ruby
[6] pry(main)> value = ''
[7] pry(main)> Win32::Registry::HKEY_LOCAL_MACHINE.open('SYSTEM\CurrentControlSet\Control\Session Manager\Environment') do |reg|t\Control\Session Manager\Environment') do |reg|

[7] pry(main)*   value = reg['PSModulePath']
[7] pry(main)* end
=> "C:\\Windows\\system32\\WindowsPowerShell\\v1.0\\Modules\\;C:\\Program Files (x86)\\Microsoft SDKs\\Windows Azure\\PowerShell\\Azure"
```

Also, note there are no config values under `HKCU\Environment` for `PSModulePath` either.


So I think that what you're doing here is right (but I might change how we get the path to `ProgramFiles`).

Thoughts?